app/ide-desktop/lib/types/globals.d.ts

```diff
@@ -19,10 +19,32 @@ interface BuildInfo {
     name: string
 }
 
+// ==========================
+// === Authentication API ===
+// ==========================
+
+/** `window.authenticationApi` is a context bridge to the main process, when we're running in an
+ * Electron context.
+ *
+ * # Safety
+ *
+ * We're assuming that the main process has exposed the `authenticationApi` context bridge (see
+ * `lib/client/src/preload.ts` for details), and that it contains the functions defined in this
+ * interface. Our app can't function if these assumptions are not met, so we're disabling the
+ * TypeScript checks for this interface when we use it. */
```

Comment:
Looks like this safety comment might not be relevant anymore?

Reply:
maybe - up to you tbh. i do believe not all of them are defined for every module, so some kind of safety warning for all the `window` augmentations (and/or global augmentations and process.env augmentations) would still be good

Reply:
maybe add comments to each variable specifying where they're defined - and potentially add comments to where they're defined, to remind anyone editing stuff there to also edit the definitions in `globals.d.ts`